### PR TITLE
Update vector_math

### PR DESCRIPTION
**Pull request by abarth**
The new version has an API change. Also, up our SDK requirements to be a
bit more realistic.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter_sprites/pubspec.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ author: Flutter Authors <flutter-dev@googlegroups.com>
 homepage: http://flutter.io
 
 dependencies:
-  box2d: '>=0.2.0 <0.3.0'
+  box2d: '>=0.3.0 <0.4.0'
```
Comment (eseidel) [inline packages/flutter_sprites/pubspec.yaml]: I'm surprised this changed?

Comment (abarth) [inline packages/flutter_sprites/pubspec.yaml]: box2d depends on vector_math as well.  We need to upgrade it to get a version that understands the new vector math.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**